store/sqlstore/channel_store.go

```diff
@@ -2825,8 +2825,9 @@ func (s SqlChannelStore) channelSearchQuery(term string, opts store.ChannelSearc
 			OrderBy("c.DisplayName, t.DisplayName").
 			Limit(uint64(limit))
 	}
-
-	if !opts.IncludeDeleted {
+	if opts.Deleted {
+		query = query.Where(sq.NotEq{"c.DeleteAt": int(0)})
+	} else if !opts.IncludeDeleted {
```

Comment:
nit: I'm not sure if this is the expected behavior. I would expect, If I ask to the api for channels, and I don't want to include the deleted ones in the list, and then I filter only by the deleted ones, I would expect and empty list, not to ignore one of my requests. I understand that makes no sense from the users perpective, but if somebody from the client is relaying on the idea of "I'm not including deleted channels, but I don't care about setting or not the Deleted filter" can end up with some weird behaviors.